apps/cowswap-frontend/src/modules/usdAmount/utils/usdcPriceLoader.ts

```diff
@@ -0,0 +1,20 @@
+import { USDC } from '@cowprotocol/common-const'
+import { FractionUtils } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Fraction } from '@uniswap/sdk-core'
+
+import { getCowProtocolNativePrice } from '../apis/getCowProtocolNativePrice'
+
+export function usdcPriceLoader(chainId: SupportedChainId): () => Promise<Fraction | null> {
+  let usdcPricePromise: Promise<number | null> | null = null
```

Comment:
But 2 parts of this app using the hook would not reuse the the promise. Only if you reuse the callback that this hook returns...

does this help? https://github.com/cowprotocol/cowswap/pull/5266

Reply:
af309d65002d